apollo-router/tests/integration/lifecycle.rs

```diff
@@ -462,3 +464,103 @@ fn test_plugin_ordering_push_trace(context: &Context, entry: String) {
         )
         .unwrap();
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_multi_pipelines() {
+    if !graph_os_enabled() {
+        eprintln!("test skipped");
+        return;
+    }
+    let mut router = IntegrationTest::builder()
+        .config(include_str!("fixtures/prometheus.router.yaml"))
+        .responder(ResponseTemplate::new(500).set_delay(Duration::from_secs(10)))
+        .build()
+        .await;
+
+    router.start().await;
+    router.assert_started().await;
+
+    let query = router.execute_default_query();
+    // Long running request 1
+    let _h1 = tokio::task::spawn(query);
+    router
+        .update_config(include_str!("fixtures/prometheus_updated.router.yaml"))
+        .await;
+
+    router.assert_reloaded().await;
+    // Long running request 2
+    let query = router.execute_default_query();
+    let _h2 = tokio::task::spawn(query);
+    let metrics = router
+        .get_metrics_response()
+        .await
+        .expect("metrics")
+        .text()
+        .await
+        .expect("metrics");
+
+    // There should be two instances of the pipeline metrics
+    let pipelines = Regex::new(r#"(?m)^apollo_router_pipelines[{].+[}] 1"#).expect("regex");
+    assert_eq!(pipelines.captures_iter(&metrics).count(), 2);
+
+    // There should be at least two connections, one active and one terminating.
+    // There may be more than one in each category because reqwest does connection pooling.
+    let terminating =
+        Regex::new(r#"(?m)^apollo_router_open_connections[{].+terminating.+[}]"#).expect("regex");
+    assert_eq!(terminating.captures_iter(&metrics).count(), 1);
+    let active =
+        Regex::new(r#"(?m)^apollo_router_open_connections[{].+active.+[}]"#).expect("regex");
+    assert_eq!(active.captures_iter(&metrics).count(), 1);
+}
+
+/// This test ensures that the rotuer will not leave pipelines hanging around
```

Comment:
```suggestion
/// This test ensures that the router will not leave pipelines hanging around
```

Reply:
we should "forward-port" this spelling bit to `dev`.  (or probably just fix it on `dev` manually inside literally any other PR, probably)

Reply:
https://github.com/apollographql/router/pull/7222